src/useSpeech/index.ts

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useEffect, useState } from 'react';
+
+type SpeechOptions = {
+	voice?: SpeechSynthesisVoice;
+	rate?: number;
+	pitch?: number;
+	volume?: number;
+	lang?: string;
+};
+
+type SpeechPlayStatus = 'init' | 'play' | 'pause' | 'end' | 'error';
+
+export type SpeechState = {
+	isPlaying: boolean;
+	status: SpeechPlayStatus;
+	errorMessage?: string;
+};
+
+/**
+ * Hook for playing text using the SpeechSynthesis API.
+ *
+ * @param {string} text - Text to play.
+ * @param {ISpeechOptions} options - Options for playing the text.
+ * @param {SpeechSynthesisVoice} [options.voice] - Voice to use for playing the text.
+ * @param {number} [options.rate] - Rate at which to play the text.
+ * @param {number} [options.pitch] - Pitch at which to play the text.
+ * @param {number} [options.volume] - Volume at which to play the text.
+ * @param {string} [options.lang] - Language of the text.
+ * @returns {SpeechState} State object representing the current state of speech synthesis.
+ */
+export const useSpeech = (text: string, options: SpeechOptions): SpeechState => {
+	const [state, setState] = useState<SpeechState>(() => {
```

Comment:
Instead of creating new function each render - you can define static one, outside hook body

Reply:
@xobotyi This change has been addressed [here](https://github.com/react-hookz/web/pull/1378/commits/06c5889add8e87fd0617263c7d65eac35d2b8879)